### ISSUE TITLE
Loader.onDuplicateKeyIgnore doesn't work on tables that don't have a PRIMARY KEY

### ISSUE THREAD

**Issue by juriad** [jOOQ/jOOQ#12741]
### Use case:

> Caused by: java.lang.IllegalStateException: ON DUPLICATE KEY IGNORE only works on tables with explicit primary keys. Table is not updatable : "_RoleMapping"

This table indeed does not have a primary key but has a multi-column unique key.

The JavaDoc on `LoaderOptionsStep.onDuplicateKeyIgnore` does not suggest that the table must have a primary key. The documentation talks specifically about unique keys and even mentions the eventuality of a table without any keys.

Moreover, the table may not even "exist". My other use-case is that I want to use the Loader API to fill a just created temporary table that jOOQ does not know anything about. So using `DSL.table(Name)` should work too.

### Possible solution you'd like to see:

There should not be any need for a primary key (or a unique key) to be defined on the table. This option translates to `InsertQuery.onDuplicateKeyIgnore` which does not have any such restriction. (I was using it but am replacing it with the Loader API.)

Loader should have the same capabilities as Insert:
```java
    @Test
    public void test() {
        ctx.transaction(trans -> {
            DSLContext t = DSL.using(trans);

            t.createTemporaryTable("abc")
                    .column(DSL.field("a", SQLDataType.INTEGER))
                    .constraint(DSL.constraint().primaryKey("a"))
                    .onCommitDrop()
                    .execute();

            int total = 0;
            total += t.insertInto(DSL.table("abc"))
                    .values(1).values(2).values(3)
                    .onDuplicateKeyIgnore().execute();
            total += t.insertInto(DSL.table("abc"))
                    .values(4).values(5).values(3)
                    .onDuplicateKeyIgnore().execute();

            System.out.println(total); // prints 5
        });

        ctx.transaction(trans -> {
            DSLContext t = DSL.using(trans);

            t.createTemporaryTable("abc")
                    .column(DSL.field("a", SQLDataType.INTEGER))
                    .constraint(DSL.constraint().primaryKey("a"))
                    .onCommitDrop()
                    .execute();

            int total = t.loadInto(DSL.table("abc"))
                    .batchAfter(3)
                    .onDuplicateKeyIgnore() // fails here
                    .loadArrays(Stream.of(1, 2, 3, 4, 5, 3).map(i -> new Object[]{i}))
                    .fields(DSL.field("a"))
                    .execute()
                    .stored();

            System.out.println(total);
        });
    }
```

### Possible workarounds:

Use Insert directly and implement batching and bulking myself; basically revert back to where I was a day ago.

### Versions:

- jOOQ: 3.15
- Java: 11
- Database (include vendor): Postgres
- OS:
- JDBC Driver (include name if inofficial driver):

**Comment by lukaseder** [comment]
Thanks for your feature request. I agree the loader flag should work like the `INSERT` flag where this is possible.  I will investigate if there's a hidden limitation specific to the `Loader` API or if this is just an oversight.

> Moreover, the table may not even "exist". My other use-case is that I want to use the Loader API to fill a just created temporary table that jOOQ does not know anything about. So using `DSL.table(Name)` should work too.

This will only work if there's native support for the feature, which there is in PostgreSQL

**Comment by lukaseder** [comment]
The feature was only introduced recently, in jOOQ 3.14: https://github.com/jOOQ/jOOQ/issues/7253

**Comment by lukaseder** [comment]
I don't see any reason to maintain the current restriction. The whole ignore logic is handled in the `InsertQueryImpl` already. The emulation using `INSERT .. SELECT` requires unique key meta data to be present (`UNIQUE` or `PRIMARY`, both work), and the `INSERT .. ON CONFLICT` emulation doesn't require any meta data.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0, 3.15.6 (https://github.com/jOOQ/jOOQ/issues/12745), and 3.14.16 (https://github.com/jOOQ/jOOQ/issues/12746).

Thanks again for your report!
